doc/wg/opentitan/case_studies/datacenter_security_model.md

```diff
@@ -0,0 +1,173 @@
+# Root-of-Trust Security Model
+
+
+### Datacenter Use Cases
+
+<p style="text-align: right">
+Author: cfrantz</p>
+
+
+<p style="text-align: right">
+Date: 2022-11-15</p>
+
+
+<p style="text-align: right">
+Status: Published</p>
+
+
+
+## Abstract
+
+This document describes the security model for a Root-of-Trust (RoT) chip as currently deployed for datacenter use cases.  The deployment described represents how the current Google proprietary root-of-trust chip (publicly known as Titan) and its firmware work.  The exact nature of the security model continues to evolve along with Google's production security requirements and the capabilities available from Root-of-Trust chips.
+
+The primary purpose of the current integrations of the Titan RoT into server products is to maintain first-instruction boot integrity and to grant the machine a valid identity in Google's production environment (colloquially known as _prod_). Titan RoTs also mitigate DoS attacks by enforcing secure firmware updates.
+
+This document concludes with some alternative concepts for firmware delivery considering Tock's process isolation model and Application ID features.
+
+
+## Background
+
+Google's Platforms team designs and contracts out the manufacture of the servers and peripherals (NICs, SSDs, custom accelerators) that comprise its production environment.  Within Google, customers (ie: product teams such as websearch, gmail, cloud, etc) purchase compute resources to execute their jobs.
+
+In order for a machine to run customer jobs, it must pass a series of health checks and acquire a cryptographic identity.  A machine which is able to present (or wield) its identity can interact with other production services such as the cluster scheduler, storage services or other internal  services (such as the web index, image recognition, etc).  A machine which cannot present its identity is only permitted to interact with a limited set of services geared towards automated or manual repair of broken machines.
+
+The Google Titan chip is a Google-designed Root-of-Trust chip.  It features a 32-bit embedded ARM Cortex-M, internal flash and RAM and crypto acceleration hardware, including a SHA hashing block, a key derivation block and a bignum accelerator.
+
+
+## How Titan is Integrated into the Datacenter
+
+The Google Titan chip is integrated into server products such that it has control over the machine's reset process and boot firmware.  Specifically, the Titan chip can both monitor and drive the application processor's reset signals and interposes on the SPI bus between the AP and its EEPROM. This design is applied to simple servers with a CPU, to BMCs, and to peripherals like NICs and accelerators.
```

Comment:
Not sure what an "AP" or "BMC" are in this context. It's probably worth spelling them out at least once.

Reply:
Added definitions.